### PR TITLE
Cv 405 add n service bus handler

### PR DESCRIPTION
**Pull request by Pauljgraham**

### CODE REVIEW

**Comment by EwanNoble** [comment]
Why netstandard 1.1? We should be improving/modernising this application

File: src/SFA.DAS.Notifications.MessageHandlers/packages.config [hunk 1]
```diff
@@ -0,0 +1,160 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (EwanNoble) [inline src/SFA.DAS.Notifications.MessageHandlers/packages.config]: That's a lot of packages

**Comment by EwanNoble** [comment]
I've found at least 4 different dotnet versions/frameworks in this, surely we can do better

File: src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard1.1</TargetFramework>
```
Comment (EwanNoble) [inline src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj]: Why netstandard 1.1?

File: src/CustomTargets.fsx [hunk 3]
```diff
@@ -13,4 +13,10 @@ Target "Dotnet Restore" (fun _ ->
     DotNetCli.Restore(fun p ->
```
Comment (EwanNoble) [inline src/CustomTargets.fsx]: This file can be deleted now Jack's changes have been merged

File: src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj (hunk 2 above)
Comment (tom-gough) [reply src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj]: .NET Framework 4.5 only supports using .net standard libraries targeting .NET Standard 1.0 or 1.1

File: src/SFA.DAS.Notifications.MessageHandlers/packages.config (hunk 1 above)
Comment (tom-gough) [reply src/SFA.DAS.Notifications.MessageHandlers/packages.config]: fixed

File: src/CustomTargets.fsx (hunk 3 above)
Comment (tom-gough) [reply src/CustomTargets.fsx]: done

File: src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj (hunk 2 above)
Comment (EwanNoble) [reply src/SFA.DAS.Notifications.Messages/SFA.DAS.Notifications.Messages.csproj]: We shouldn't have any net framework 4.5 projects either 😁

File: src/SFA.DAS.Notifications.Infrastructure/DependencyResolution/MessagePolicy.cs [hunk 4]
```diff
@@ -32,7 +32,7 @@ protected override void apply(Type pluginType, IConfiguredInstance instance)
             var environment = Environment.GetEnvironmentVariable("DASENV");
             if (string.IsNullOrEmpty(environment))
             {
-                environment = ConfigurationManager.AppSettings["EnvironmentName"];
+                environment = CloudConfigurationManager.GetSetting("EnvironmentName");
```
Comment (EwanNoble) [inline src/SFA.DAS.Notifications.Infrastructure/DependencyResolution/MessagePolicy.cs]: Noooo

File: src/SFA.DAS.Notifications.Api/packages.config [hunk 5]
```diff
@@ -1,63 +1,65 @@
...
-  <package id="StructureMap.MVC5" version="3.0.4.125" targetFramework="net45" />
-  <package id="structuremap.web" version="3.0.0.108" targetFramework="net45" />
-  <package id="StructureMap.WebApi2" version="3.0.4.125" targetFramework="net45" />
-  <package id="System.IdentityModel.Tokens.Jwt" version="4.0.2.206221351" targetFramework="net45" />
-  <package id="System.Spatial" version="5.7.0" targetFramework="net45" />
-  <package id="WebActivatorEx" version="2.0.5" targetFramework="net45" />
-  <package id="WindowsAzure.ServiceBus" version="3.3.2" targetFramework="net45" />
-  <package id="WindowsAzure.Storage" version="6.2.0" targetFramework="net45" />
+  <package id="CommonServiceLocator" version="1.3" targetFramework="net46" />
+  <package id="FAKEBuildScript" version="1.2.7" targetFramework="net462" />
```
Comment (EwanNoble) [inline src/SFA.DAS.Notifications.Api/packages.config]: This one can be removed now 🎉

File: src/SFA.DAS.Notifications.Infrastructure/DependencyResolution/MessagePolicy.cs (hunk 4 above)
Comment (tom-gough) [reply src/SFA.DAS.Notifications.Infrastructure/DependencyResolution/MessagePolicy.cs]: don't worry it's gone :D

**Comment by CodeWolf86** [comment]
Notifications Status Update for conversion to .Net Core + App Service.
- All projects have been converted to .Net Core
- The build pipeline has been completed as has the release pipeline initially for AT
- There is a corresponding configuration branch in das-employerconfig which has the updated Notification Config and has been tidied up.
- The solution now has a database project added to allow for the SQLPersistance of NServiceBus which was missing.
- The project has been tidied up to make use of the AutoConfiguration so other services have been taken out as a result.
- The current branches for Notifications and Config has been deployed to AT and needs to be tested. 
- It works locally to the point it now sends an email to the email service which is rejected by the Gov Notification Api
- Unsure if the Notification API actually works in Test environments so the next steps may well be to add in SMTP Configuration into local configuration storage and try running it. 
- Then next steps will be to test in a test environment, either a test harness pointing at the shared nservicebus connection string and then a local email service temporarily in test with a custom SMTP setup or confirming if Gov Notify actually works

**Pull request merged**